pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -449,9 +450,17 @@ func ExtractMsgDetails(msg *dns.Msg) (qname string, responseIPs []net.IP, TTL ui
 			rrName = strings.ToLower(ans.Target)
 			CNAMEs = append(CNAMEs, ans.Target)
 		}
+		types = append(types, ans.Header().Rrtype)
 	}
 
-	return qname, responseIPs, TTL, CNAMEs, nil
+	var dnsMsgType int
+	if msg.Response {
+		dnsMsgType = msg.Rcode
+	} else {
+		dnsMsgType = msg.Opcode
+	}
```

Comment:
If we are going to include both, I'd prefer to avoid interpreting this data here and just passing both out to the accesslog. In particular, we should never see non-query requests but it is possible (and will happen now that I said that). The RCode is far more important with debugging etc. so ensuring that it is always clearly represented wins out (for me). If I had to choose, I'd pick the rcode.